examples/permissioned-token/test/common.js

```diff
@@ -0,0 +1,380 @@
+      recipient: to.address,
+      assetID: assetIndex,
+      revocationTarget: from.address,
+      amount: amount,
+      lsig: lsig,
+      payFlags: { totalFee: 1000 }
+    },
+    {
+      type: types.TransactionType.TransferAlgo,
+      sign: types.SignType.SecretKey,
+      fromAccount: from.account,
+      toAccountAddr: lsig.address(),
+      amountMicroAlgos: 1000,
+      payFlags: { totalFee: 1000 }
+    },
+    {
+      type: types.TransactionType.CallNoOpSSC,
+      sign: types.SignType.SecretKey,
+      fromAccount: from.account,
+      appId: permissionsAppId,
+      payFlags: { totalFee: 1000 },
+      appArgs: [TRANSFER],
+      accounts: [from.address, to.address]
+    }
+  ];
+  runtime.executeTx(txGroup);
+}
+
+/**
+ * Performs Opt-Out operation
+ * @param runtime RuntimeEnv
+ * @param manager ASA Manager Account
+ * @param opAccount Account to be Opted-Out
+ * @param assetIndex ASA ID
+ */
+function optOut (runtime, manager, opAccount, assetIndex) {
+  const optOutParams = {
+    type: types.TransactionType.TransferAsset,
+    sign: types.SignType.SecretKey,
+    fromAccount: opAccount.account,
+    toAccountAddr: opAccount.address,
+    assetID: assetIndex,
+    amount: 0,
+    payFlags: { totalFee: 1000, closeRemainderTo: manager.address }
+  };
+  runtime.executeTx(optOutParams);
+}
+
+/**
+ * Force Transfer Tokens
+ * @param runtime RuntimeEnv Instance
+ * @param from Sender Account
+ * @param to Receiver Account
+ * @param amount Amount
+ * @param assetIndex ASA ID
+ * @param controllerAppID Controller App ID
+ * @param permissionsAppId Permissions App ID
+ * @param lsig Clawback LSig
+ * @param manager ASA Manager Account
+ */
+function forceTransfer (
+  runtime, from, to, amount, assetIndex, controllerAppID, permissionsAppId, lsig, manager) {
+  const txGroup = [
+    {
+      type: types.TransactionType.CallNoOpSSC,
+      sign: types.SignType.SecretKey,
+      fromAccount: manager.account,
+      appId: controllerAppID,
+      payFlags: { totalFee: 1000 },
+      appArgs: ['str:force_transfer'],
+      accounts: [to.address],
+      foreignAssets: [assetIndex]
+    },
+    {
+      type: types.TransactionType.RevokeAsset,
+      sign: types.SignType.LogicSignature,
+      fromAccountAddr: lsig.address(),
+      recipient: to.address,
+      assetID: assetIndex,
+      revocationTarget: from.address,
+      amount: amount,
+      lsig: lsig,
+      payFlags: { totalFee: 1000 }
+    },
+    {
+      type: types.TransactionType.TransferAlgo,
+      sign: types.SignType.SecretKey,
+      fromAccount: manager.account,
+      toAccountAddr: lsig.address(),
+      amountMicroAlgos: 1000,
+      payFlags: { totalFee: 1000 }
+    },
+    {
+      type: types.TransactionType.CallNoOpSSC,
+      sign: types.SignType.SecretKey,
+      fromAccount: manager.account,
+      appId: permissionsAppId,
+      payFlags: { totalFee: 1000 },
+      appArgs: [TRANSFER],
+      accounts: [from.address, to.address]
+    }
```

Comment:
We should also add a test where we don't pass permissions app call in `force_transfer`

Reply:
I tried making a force transfer where the receiver is asset reserve address, but the compiled TEAL is throwing error in that case.